src/Functions/DateTimeTransforms.h

```diff
@@ -175,6 +175,28 @@ struct ToStartOfMinuteImpl
     using FactorTransform = ZeroTransform;
 };
 
+struct ToStartOfSecondImpl
+{
+    static constexpr auto name = "toStartOfSecond";
+
+    static inline DateTime64 execute(const DateTime64 & datetime64, Int64 scale_multiplier, const DateLUTImpl &)
+    {
+        const auto fractional = DecimalUtils::getFractionalPartWithScaleMultiplier(datetime64, scale_multiplier);
+        return datetime64 - fractional;
```

Comment:
It does not look correct for negative values (that will be introduced in another PR).

Example:

value is -0.3
fractional part that you calculate is 0.3
value - fractional = -0.6

But if I understand correctly, you want to round down to -1.0 (towards minus infinity).

Reply:
nice catch, fixed that